### ISSUE TITLE
New release 2.2.1?

### ISSUE THREAD

**Issue by olebole** [iraf-community/pyraf#139]
Since a number of important fixes are now integrated, I created a new release candidate [2.2.1RC1](https://github.com/iraf-community/pyraf/releases/tag/v2.2.1rc1) last week. The version can also be downloaded [from PyPI](https://pypi.org/project/pyraf/2.2.1rc1/). Please use this issue to make comment on it (or create a new issue), or to indicate green light if you like.

@jehturner do you see something that should be fixed before releasing?

**Comment by jehturner** [comment]
I was just having another quick look at the dependencies, WRT the conda recipe we're using. Is this import still needed?
https://github.com/iraf-community/pyraf/blob/b763a51fcfef1ee604f8fc145a739d9f059e6006/pyraf/tests/utils.py#L4
Maybe best get rid of that if not, so it doesn't break in future? More generally, it looks like `astropy` is only required for running tests (using `io.fits`) & building docs (html_theme), making it semi-optional (indeed you only have it as a testing requirement in `setup.cfg`).

Other than that, I believe we should remove `d2to1`, `stsci.distutils` & `stsci.tools` from the conda recipe and add `configobj >=5`. That recipe also lists `ipython` & `matplotlib` (+ some `pyobjc` stuff on macos), which should be optional but are probably useful to pull in (& that's up to STScI anyway if they are going to keep building it).

**Comment by olebole** [comment]
@jehturner that all looks reasonable. If you like, you can make a simple last-minute-PR for this; otherwise I will do tomorrow.
For astroconda, I have no opinion -- maybe ping someone of them so that they are aware?

**Comment by jehturner** [comment]
See #140.

I'll put in an AstroConda PR and talk to them separately, as there are a few things that I need to catch up with them on.

**Comment by jehturner** [comment]
This looks good to me on Linux. I built an `rc1` conda package, installed it with a recent set of packages that users would get, manually patched in the import removal from #140 and re-ran our tests with the same internal Gemini IRAF version as before, getting only the same 14 unrelated test failures out of 932 as previously. On Apple, `pyraf` failed to run at all (even from PyPI), producing a `libmkl` error, but I think that's probably just a problem with the latest Anaconda Python packages on an old machine/OS; I'll need to look into more tomorrow.

**Comment by jehturner** [comment]
I had rather a painful time building a conda package for Apple, due to a problem with Anaconda dependencies on the available build machine, but no need to bore you with that one... Unfortunately, after eventually getting PyRAF installed properly, it turns out that #99 is still a problem on MacOS (actually, `pyraf` won't start at all when a default `login.cl` has been created), so it's not usable :slightly_frowning_face:. I confirmed that the same is also true when using your PyPI build, FWIW. Sorry, our main set of Gemini IRAF tests only runs on Linux... Would it be possible to apply the same fix for 32-bit `macosx` as `linux` please? :pray:

**Comment by olebole** [comment]
Which fix do you mean? This one? https://github.com/iraf-community/pyraf/blob/abff01b45326537cfca42707c35b0cfe8fdec099/pyraf/iraffunctions.py#L181-L189

Sure, we can just add macosx here… I'd however really recommend to (also) apply the fix in IRAF's `/unix/hlib/libc/kernel.h`
```C
#define        ADDR_TO_LOC(addr)       (((unsigned XINT)((XCHAR *)(addr)))>>(sizeof(XCHAR)-1))
```
since this fixes the cause and will avoid problems if people use the traditional cl/ecl.

**Comment by olebole** [comment]
Yes, this was the fix for Linux. I think that this line is similar to setting the stack size to unlimited -- but maybe this isn't automatically propagated to Python childs?
Whatever, if it works for you, I think it is fine :grin:

**Comment by jehturner** [comment]
Those were my thoughts too...

**Comment by jehturner** [comment]
The `Not a legal pipe record` on `macosx` has gone away after #141, thanks! I forgot to mention that I'm also seeing `No graphics/display possible for this session`, unexpectedly, but I'll just double check whether I have messed up X11 forwarding or something.

**Comment by jehturner** [comment]
My PyRAF display problem turned out to be just another problem with the Anaconda stack on that machine, but it prevented me from testing the build there properly (& I couldn't test it on an up-to-date machine because 32-bit IRAF won't run on 10.15+). However, my colleague reports that my build of [workaround-old-macos32](https://github.com/iraf-community/pyraf/tree/workaround-old-macos32) worked well for her on 10.13, from a few quick tests, and the included `pyraf/tests/` passed, so I think we're OK :slightly_smiling_face:. We have only tried it on Python 3.7 so far (which version(s) are you running it with?).

**Comment by olebole** [comment]
I run it 
* locally: 
  - Python 3.9.11, Debian testing, 
  - Python 3.8.2, macOS 11.6 (in a VirtualBox),
* in Github CI
  -  Python 3.8.2, Ubuntu 20.04,
  -  Python 3.7, Ubuntu 18.04,
  -  Python 3.9.10, macOS 10.15.

All with IRAF community versions, all 64 bit only. Tests with a graphics output only locally. If you have a simple setup for Github actions, we could extend that to a test with AstroConda.

**Comment by jehturner** [comment]
Thanks for the offer. We don't really have an easy way to integrate our tests, but I'll bear that in mind.

I've confirmed that our regression tests also pass under Anaconda's Python 3.9 on CentOS 7, with no difference from 3.7 (and I can also run that build and produce plots on ALMA/CentOS 8, as you'd expect).

**Comment by olebole** [comment]
I just tagged a [new version 2.2.1](https://github.com/iraf-community/pyraf/releases/tag/v2.2.1) and [uploaded it to PyPI](https://pypi.org/project/pyraf/2.2.1/), so this issue can be closed.
